timer.go

```diff
@@ -0,0 +1,42 @@
+package libtime
+
+import (
+	"time"
+)
+
+// StopFunc is used to stop a time.Timer.
+type StopFunc func()
+
+// SafeTimer creates a time.Timer and a StopFunc, forcing the caller to deal
+// with the potential goroutine leak. Encourages safe use of a time.Timer in a
+// select statement, but without the overhead of a context.Context.
+//
+// Typical usage:
+//
+//    t, stop := libtime.SafeTimer(interval)
+//    defer stop()
+//    for {
+//      select {
+//        case <- t.C:
+//          foo()
+//        case <- otherC :
+//          return
+//      }
+//    }
+//
+// Does not panic if duration is <= 0, instead assuming the smallest positive value.
+func SafeTimer(duration time.Duration) (*time.Timer, StopFunc) {
+	if duration <= 0 {
+		// Do not panic on zero value, and instead use the smallest positive value
+		// which is in practice identical to the behavior of time.After(0), which
+		// this helper replaces.
+		//
+		// https://go.dev/play/p/EIkm9MsPbHY
+		duration = 1
+	}
```

Comment:
I peeked at `time.NewTimer` and it appears to have this same logic baked in. Let's drop this check?

Reply:
Agh you're right! I was thinking of  `time.NewTicker` 

fixed